Dockerfile

```diff
@@ -0,0 +1,73 @@
+# https://yogin16.github.io/2018/08/30/grpc-server-dockerize/
+# https://stackoverflow.com/questions/60014845/how-to-install-oracle-jdk11-in-alpine-linux-docker-image/68459967#68459967
+
+FROM eclipse-temurin:11 as build
+
+
+# This is a build image - no need to update anything, just build.
+RUN apt-get update && \
+    apt-get upgrade && \
+    apt-get install --yes tzdata wget unzip bash
+
+## protoc and gRPC helper binaries included with GRPC and ProtoBuf Gradle plugins
+## are built against glibc. But Alpine linux uses musl - and this leads to all
+## sorts of cryptic errors while trying to build .proto and gRPC stubs.
+## The package `gcompat` adds a glibc compatibility layer that just solves the
+## problem with minimal fuss. See
+## * https://wiki.alpinelinux.org/wiki/Running_glibc_programs
+## * https://github.com/google/protobuf-gradle-plugin/issues/265
+## We also add protobuf package as there is a lib  dependency on it.
+#RUN apk add --no-cache --update gcompat protobuf \
+#    && rm -rf /var/cache/apk/*
+
+#
+# We will install things in opt. Create it already if it not there.
+#
+
+RUN mkdir -vp /opt/
+
+#
+#  Install Gradle
+#
+# We install gradle directly from its sources to ensure we use a recent
+#  version and to avoid its packager pulling unecessary dependencies.
+ENV GRADLE_VERSION 7.4
+RUN mkdir /tmp/gradle-dl
+WORKDIR /tmp/gradle-dl
+RUN wget https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip \
+    && unzip -d /opt/ gradle-${GRADLE_VERSION}-bin.zip \
+    && rm -rfv /tmp/gradle-dl
+ENV PATH="/opt/gradle-${GRADLE_VERSION}/bin:${PATH}"
+
+#
+# Build the project
+#
+ENV PROJECT_DIR=/opt/statestore-component
+RUN mkdir -vp ${PROJECT_DIR}
+COPY ./ ${PROJECT_DIR}
+WORKDIR ${PROJECT_DIR}
+RUN gradle installDist
+
+#
+# Build complete. Now setup only the the runtime environment
+#
+
+FROM eclipse-temurin:11-jre
+
+RUN mkdir -vp /opt/
+ENV PROJECT_DIR=/opt/statestore-component
+COPY --from=build ${PROJECT_DIR} ${PROJECT_DIR}
+
+ENV PATH="/opt/statestore-component/build/install/DaprPluggableComponent-Java/bin/:${PATH}"
+
+# Copy umask wrapper. We need it to ensure the Unix Socket Domain created by
+# This container is readable and writable by the outside Dapr process
+
+COPY bin/umask_entrypoint_wrapper.sh /usr/local/bin
```

Comment:
should the sdk code handle this?

I think it's a pre-requisite to work on UNIX-like OS's,

Having this outside the sdk is certainly an option, however we should let clear for the component implementer that they must ensure the container process is able to create files allowed to be r/w for other processes

On the components-go-sdk I chose to have this inside the sdk rather than handled by the implementer. I found it better because I can switch between unix/non-unix using build flags =>

https://github.com/dapr-sandbox/components-go-sdk/blob/main/socket_unix.go#L22
https://github.com/dapr-sandbox/components-go-sdk/blob/main/socket_unix.go#L1

A non-obvious option is provide base images for the sdks - which could be a good idea for the long-term, but for now, I'd rather we handle it ourselves

Reply:
The problem is that, if I go it right, Go has provides APIs for `umask` out of the box. That is not the case with Java: its SDK has no support for `umask` and to add it one has to depend on a 3rd party library that used JNI to invoke this call. So, in other words, libraty that must also include `.so` code and other platorm and os dependent code.

Is this so problematic? Not necessarily: this is what gRPC does.  And that was the reason we needed to include a glibc compatibility layer when the base image was Alpine linux (as Alpine does not use glibc). The problem is: is this really that much better than using a tiny helper shell script that will run just fine in MacOS, Linux, FreeBSD and whatever?

I will investigate the workarounds as pointed out on https://stackoverflow.com/questions/3175303/how-can-i-set-the-umask-from-within-java

Reply:
Addressed this on https://github.com/dapr-sandbox/components-java-sdk/pull/7#discussion_r985002876